source/agora/test/FutureEnrollment.d

```diff
@@ -0,0 +1,114 @@
+/*******************************************************************************
+
+    Contains tests for accepting "future" enrollments
+
+    Copyright:
+        Copyright (c) 2019-2021 BOSAGORA Foundation
+        All rights reserved.
+
+    License:
+        MIT License. See LICENSE for details.
+
+*******************************************************************************/
+
+module agora.test.FutureEnrollment;
+
+version (unittest):
+
+import agora.api.FullNode;
+import agora.common.Amount;
+import agora.common.Config;
+import agora.consensus.data.Block;
+import agora.consensus.data.Enrollment;
+import agora.consensus.data.Transaction;
+import agora.crypto.Key;
+import agora.test.Base;
+
+/// Situation: A slow validator is a block behind the latest height(19)
+///     where other nodes have a block that contains a frozen UTXO for
+///     the enrollment for the outsider. The slow validator only has
+///     the `tx` in the pool for the frozen UTXO. At that time, the
+///     outsider tries to enroll and the bad validator receives the
+///     enrollment.
```

Comment:
Why is the validator referred to as "bad" or "slow" ? It isn't either.
Also, the fix checks the TxPool, which is not really tested here IIUC.
It'd be better to let everyone re-enroll, but have an outsider send a freezing transaction, then an enrollment, and check that it correctly gets externalized.